src/step-function.ts

```diff
@@ -153,18 +132,25 @@ export namespace $SFN {
    * @param props configure the maxConcurrency
    * @param callbackfn function to process each item
    */
-  // @ts-ignore
-  export function forEach<T>(
-    array: T[],
-    props: {
-      maxConcurrency: number;
+  export const forEach = makeIntegration<forEach>({
```

Comment:
I worry about how these function signatures have been destroyed. If a developer is browsing through trying to learn what they can call, won't they be like "wtf"? Also, can we still document the functions so that it renders properly in both the IDE and generated docs?

Reply:
Document the forEach and map parameters. Do they properly make it to the IDE? What do the generated docs look like?

Reply:
You were right, after messing it, union anonymous method signatures were not correctly documented. Simple signatures like `getItem` were fine, they showed the ts-docs, but the union/overload types did not.

This seems to work

```ts
  interface ForEach {
    /**
     * Process each item in an {@link array} in parallel and run with the default maxConcurrency.
     *
     * Example:
     * ```ts
     * new ExpressStepFunction(this, "F"} (items: string[]) => {
     *   $SFN.forEach(items, { maxConcurrency: 2 }, item => task(item));
     * });
     * ```
     *
     * @param array the list of items to process
     * @param callbackfn function to process each item
     */
    <T>(
      array: T[],
      callbackfn: (item: T, index: number, array: T[]) => void
    ): void;
    /**
     * Process each item in an {@link array} in parallel and run with the default maxConcurrency.
     *
     * Example:
     * ```ts
     * new ExpressStepFunction(this, "F"} (items: string[]) => {
     *   $SFN.forEach(items, { maxConcurrency: 2 }, item => task(item));
     * });
     * ```
     *
     * @param array the list of items to process
     * @param props configure the maxConcurrency
     * @param callbackfn function to process each item
     */
    <T>(
      array: T[],
      props: {
        maxConcurrency: number;
      },
      callbackfn: (item: T, index: number, array: T[]) => void
    ): void;
  }

  export const forEach = makeIntegration<ForEach>({
    ...sfnIntegrationBase("forEach"),
    asl(call, context) {
      return mapOrForEach(call, context);
    },
  });
```

In fact it is better than before because now each of the signatures have their own params documented separately. 

I am unsure how to generate the docs.

Reply:
![image](https://user-images.githubusercontent.com/289213/165374439-3ae04ef1-643f-4599-9ab1-ec34c200d834.png)
![image](https://user-images.githubusercontent.com/289213/165374591-193eb628-1489-41e2-b0ae-1d13bf1b2662.png)